spec/services/preview_service_spec.rb

```diff
@@ -0,0 +1,117 @@
+      it "uploads the assets to Asset Manager" do
+        image_revision = create(:image_revision)
+        edition = create(:edition, image_revisions: [image_revision])
+
+        request = stub_asset_manager_receives_an_asset
+        PreviewService.new(edition).create_preview
+
+        expect(request).to have_been_requested.at_least_once
+        expect(image_revision.assets.map(&:state).uniq).to match(%w[draft])
+      end
+    end
+
+    context "when there are assets that are on Asset Manager" do
+      it "doesn't upload the assets to asset manager" do
+        image_revision1 = create(:image_revision, :on_asset_manager)
+        image_revision2 = create(:image_revision, :on_asset_manager, state: :live)
+        edition = create(:edition, image_revisions: [image_revision1, image_revision2])
+
+        request = stub_asset_manager_receives_an_asset
+        PreviewService.new(edition).create_preview
+
+        expect(request).not_to have_been_requested
+        expect(image_revision1.assets.map(&:state).uniq).to match(%w[draft])
+        expect(image_revision2.assets.map(&:state).uniq).to match(%w[live])
+      end
+    end
+
+    context "when Publishing API is down" do
+      it "marks the edition as revision not synced" do
+        stub_publishing_api_isnt_available
+        edition = create(:edition, revision_synced: true)
+
+        expect { PreviewService.new(edition).create_preview }
+          .to raise_error(GdsApi::BaseError)
+        expect(edition.revision_synced).to be(false)
+      end
+    end
+
+    context "when there are images and Asset Manager is down" do
+      it "marks the edition as revision not synced" do
+        stub_asset_manager_isnt_available
+        image_revision = create(:image_revision)
+        edition = create(:edition, image_revisions: [image_revision], revision_synced: true)
+
+        expect { PreviewService.new(edition).create_preview }
+          .to raise_error(GdsApi::BaseError)
+        expect(edition.revision_synced).to be(false)
+      end
+    end
+  end
+
+  describe "#try_create_preview" do
+    it "updates the Publishing API" do
+      edition = create(:edition)
+      request = stub_publishing_api_put_content(edition.content_id, {})
+      PreviewService.new(edition).create_preview
+      expect(request).to have_been_requested
+    end
+
+    context "when an external service is down" do
+      it "swallows the exception and marks the edition as revision not synced" do
+        stub_publishing_api_isnt_available
+        edition = create(:edition, revision_synced: true)
+
+        expect { PreviewService.new(edition).try_create_preview }
+          .not_to raise_error
+        expect(edition.revision_synced).to be(false)
+      end
+    end
+
+    context "when there are pre-preview issues" do
+      let(:edition) { create(:edition, title: "", revision_synced: true) }
+
+      it "marks the edition as revision not synced and doesn't sent to the Publishing API" do
```

Comment:
I think this would be a bit easier to read it the 'and's were separate 'it's. We seem to do that pretty consistently everywhere else. What do you think?

Reply:
Agreed have amended